packages/chaire-lib-common/src/services/routing/RoutingResult.ts

```diff
@@ -6,36 +6,63 @@
  */
 import _get from 'lodash/get';
 
-import Preferences from 'chaire-lib-common/lib/config/Preferences';
-import { Route } from 'chaire-lib-common/lib/services/routing/RoutingService';
-import { RoutingOrTransitMode, RoutingMode } from 'chaire-lib-common/lib/config/routingModes';
-import TrError, { ErrorMessage } from 'chaire-lib-common/lib/utils/TrError';
-import { TrRoutingRoute } from 'chaire-lib-common/lib/services/trRouting/TrRoutingService';
+import Preferences from '../../config/Preferences';
+import { Route } from './RoutingService';
+import { RoutingOrTransitMode, RoutingMode } from '../../config/routingModes';
+import TrError, { ErrorMessage } from '../../utils/TrError';
+import { TrRoutingRoute } from '../trRouting/TrRoutingService';
+
+export const pathIsRoute = (path: Route | TrRoutingRoute | undefined): path is Route => {
+    return typeof (path as any).distance === 'number';
+};
```

Comment:
Some transit route have no distance calculated if the shape of the path is not provided in the GTFS. Should we chack for travel time instead?

Reply:
OK, but could we have a route result with only a travel time/duration and no distance? A ferry route by instance?

Reply:
I would say that routing engine wil alsways return a duration, but not necessarly a distance

Reply:
I definitely cannot know about "always", either for the duration or the distance. But osrm's doc says it returns a distance and we, when we handle the routing results, always set it and the field is mandatory. Maybe it's 0, or -1, but it's there and a number!

Reply:
OK then !